app/pages/SearchResultsPage/SearchResultsPage.tsx

```diff
@@ -34,22 +36,64 @@ interface SearchState extends ParsedQs {
 
 /** Wrapper component that handles state management, URL parsing, and external API requests. */
 export const SearchResultsPage = () => {
+  const [cookies] = useCookies(["googtrans"]);
+  const history = useHistory();
+  const { search } = useLocation();
   const { userLocation } = useAppContext();
   const [lastPush, setLastPush] = useState(Date.now());
-  const { search } = useLocation();
   const [expandList, setExpandList] = useState(false);
-
-  const searchState: SearchState = useMemo(
-    () => qs.parse(search.slice(1)),
-    [search]
-  );
+  const [searchState, setSearchState] = useState<SearchState>({});
   const [searchRadius, setSearchRadius] = useState(
     searchState?.configure?.aroundRadius ?? "all"
   );
+  // In cases where we translate a non-English query into English (to query Alglolia), we use this
+  // state var to cache the user's orginal, untranslated query to display on the results page
+  const [untranslatedQuery, setUntranslatedQuery] = useState("");
+  const [searchStateResolved, setSearchStateResolved] = useState(false);
+
+  useEffect(() => {
+    const queryParams = qs.parse(search.slice(1));
```

Comment:
Could you remind me how the flow of data works on the Search Results page, specifically how the initial page URL, the internal React State, and the Algolia InstantSearch widget works?

I forget who was in charge of updating the URL's query parameters, but I just wanted to make sure that we are properly updating the SearchState here when the user interacts with the Algolia widgets like changing filters or changing the search query. I'm a little worried that this `qs.parse()` code will run every time this effect runs, which I think is different from what we had before, where I think it was only running once on initial page load.

Reply:
Update to this comment, specifically ```Toggling a filter triggers a search via the InstantSearch widget, which then triggers the onSearchStateChange callback, which we use to sync the URL and our component's state with Algolia's InstantSearch widget.```

Actually, my order is a bit wrong. The correct order, as I understand it (now), is: Toggling a filter does the following: 

1. Triggers the onSearchStateChange event, which passes a `nextSearchState` param to its callback function
2. We check the nextSearchState value passed by the event and update our URL accordingly 
3. We push the URL onto the history stack which causes our page to re-render
4. During the render, we update the searchState React state value using the new URL params
5. This in turn triggers an Algolia search, I'm assuming because it contains a reference to the searchState state prop

Reply:
I should hopefully have this finished up by tomorrow

Reply:
Oh geez, thanks for that update. I don't think I actually realized that our page worked that way, and I probably thought it was something closer to your original comment. Thanks again for digging into this deeper!